decentralizediso20022/src/main/java/com/centiglobe/decentralizediso20022/config/DecentralizedISO20022Config.java

```diff
@@ -64,26 +73,31 @@
      * Generates a secure <code>ReactorClientHttpConnector</code>.
      * 
      * @return the secure <code>ReactorClientHttpConnector</code>
-     * @throws KeyStoreException                  if keystore is the problem
+     * @throws KeyStoreException                  if the keystore is the problem
      * @throws NoSuchAlgorithmException           if the algorithm does not exist
      * @throws CertificateException               if there is a problem with the
      *                                            certificate
      * @throws InvalidAlgorithmParameterException if the algorithm is invalid or
      *                                            inappropriate
-     * @throws IOException                        if any problems occure while
+     * @throws IOException                        if any problems occur while
      *                                            reading the truststore
+     * @throws UnrecoverableKeyException          if the keys could not be recovered
+     *                                            from the keystore
      */
     @Bean
     public ReactorClientHttpConnector getReactorClientConfig() throws KeyStoreException, NoSuchAlgorithmException,
-            CertificateException, InvalidAlgorithmParameterException, IOException {
-        // extract the trust store path relative to the resource folder.
+            CertificateException, InvalidAlgorithmParameterException, IOException, UnrecoverableKeyException {
+        // extract the key store paths relative to the resource folder.
+        String keystore = new File(KEY_STORE).getName();
         String truststore = new File(TRUST_STORE).getName();
-
-        TrustManagerFactory tm = createTrustManager(truststore, TRUST_PASS);
-
-        SSLFactory sslFactory = SSLFactory.builder().withTrustMaterial(tm).build();
-
-        SslContext sslContext = NettySslUtils.forClient(sslFactory).build();
+        
+        KeyManagerFactory kmf = HTTPSFactory.createKeyManager(keystore, KEY_PASS);
+        TrustManagerFactory tmf = HTTPSFactory.createTrustManager(truststore, TRUST_PASS);
+
+        SslContext sslContext = SslContextBuilder.forClient()
+            .keyManager(kmf)
+            .trustManager(tmf)
+            .build();

```

Comment:
I don't think we are using anything from the `sslcontext-kickstart-for-netty` dependency anymore, if that's the case we might want to remove it from `pom.xml`.

This solution looks better than the old one 👍

Reply:
Thanks, I removed the netty dependency

Reply:
Did you run the tests to make sure nothing is broken?

Reply:
I tested manually using Postman and the tests all pass.